config/config.go

```diff
@@ -936,15 +941,23 @@ type PackageConfig struct {
 	// Types of the Package. If not specified it is assumed to be ml_model.
 	Type PackageType `json:"type,omitempty"`
 
+	Status *AppValidationStatus `json:"status,omitempty"`
+
 	alreadyValidated bool
 	cachedErr        error
 }
 
 // Validate package config is valid.
 func (p *PackageConfig) Validate(path string) error {
+
 	if p.alreadyValidated {
 		return p.cachedErr
 	}
+
+	if p.Status != nil {
+		return utils.NewConfigValidationError(path, errors.New(p.Status.Error))
```

Comment:
@benjirewis just wanted to check something with you on this PR. Should we be storing this as a cached error? This new error is essentially a validation error from app to show that something is wrong with the module.

Reply:
I guess I would yeah. `Validate`ion usually happens multiple times for a single package during a reloading of the config/reconfiguration of the robot. Eric put in this caching system to stop running the full `Validate` multiple times for every resource/module/package. I would `p.alreadyValidated = true; p.cachedErr = utils.NewConfigValidationError(...); return p.cachedErr` here or something of the like. It's not the end of the world if you don't, but you'll be checking that `p.Status` field and creating a new error multiple times unnecessarily.

Reply:
okay sounds good I changed based on the convo ^